ibm/resource_ibm_satellite_cluster.go

```diff
@@ -513,13 +513,14 @@ func resourceIBMSatelliteClusterRead(d *schema.ResourceData, meta interface{}) e
 	}
 
 	var zones = make([]map[string]interface{}, 0)
-	for _, zone := range workerPool.Zones {
-		zoneInfo := map[string]interface{}{
-			"id": *zone.ID,
+	if res, ok := d.GetOk("zones"); ok {
+		zoneList := res.(*schema.Set).List()
```

Comment:
why are you reading d.GetOk("zones")

Reply:
Removed GetOk method and set the zones parameter in the imported section.